### PR TITLE
Add set club stage command

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

**Comment by MaxWofford** [comment]
The things to test are:

1. Set active club to dormant, then back to active
1. Set club to active again, and it should warn that it already was in that stage
1. Link your leader to 2 clubs, then set the stage of one of the clubs
1. Link your leader to 2 clubs, then try pressing the club select buttons from a different slack account

**Comment by MaxWofford** [comment]
Forgot the admin command tests:

1. Normal flow of setting one club with a valid streak_key
1. Try setting a club with an invalid streak_key
1. Try setting a club to an invalid stage
1. Try starting the command from a non-admin account

**Review by paked — CHANGES_REQUESTED** [review]
A few things:

- We need to record the date in which clubs moving to dormant are going to come back. Otherwise for all we know they're just dead. Check out the `check_in.rb` file for a reasonable guide on how to do this.
- We need to either consistently be using `club.make_{active,dormant}` or just setting the stage directly with a string (`club.stage_key = "XYZ" && club.save`)
- The code here is great. Good job.

**Comment by paked** [comment]
@MaxWofford: Can you fix the tests?

File: app/models/club.rb [hunk 1]
```diff
@@ -45,6 +47,27 @@ class Club < ApplicationRecord
...
+  def notify_of_stage_change
+    poc = point_of_contact
+
+    return if poc.nil?
+
+    msg = CopyService.new('models/club', {}).get_copy("stage.#{stage_name}")
+
+    return if msg.nil?
+
+    SlackClient::Chat.send_msg(
```
Comment (paked) [inline app/models/club.rb]: Should we turn this into a conversation?

File: app/models/hackbot/interactions/set_club.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          attachments: [
+            actions: actions
+          ]
+        )
+
+        :wait_for_club_select
+      end
+
+      # rubocop:disable Metrics/AbcSize
+      def set_single_club
```
Comment (paked) [inline app/models/hackbot/interactions/set_club.rb]: Want to bring up my concerns here again:

This is a different method of setting the stage key than we used in the `CheckIn` class. In the `CheckIn` class we used the `make_dormant` and `make_active` helper functions which I wrote.

Comment (MaxWofford) [inline app/models/hackbot/interactions/set_club.rb]: @paked I think they have different functions. The set_club interaction is set up to quickly work with any stage in Streak.

**Comment by paked** [comment]
@MaxWofford: Blocking on you to make the changes we discussed in-person.

**Comment by MaxWofford** [comment]
@paked changed!

**Comment by paked** [comment]
@MaxWofford: What are we doing here?

Worth closing?

**Comment by maxwofford** [comment]
Closing this because it's not relevant to our current work. Might reopen later if we decide we need this command.

**Pull request closed without merging**